src/ui/common/src/components/workflows/nodes/Node.tsx

```diff
@@ -41,90 +51,148 @@ export const Node: React.FC<Props> = ({
     execState = workflowState.artifactResults[data.nodeId]?.result?.exec_state;
   }
 
+  if (!execState || !execState.status) {
+    return null;
+  }
+
   const textColor = selected
     ? theme.palette.DarkContrast50
     : theme.palette.DarkContrast;
   const borderColor = textColor;
 
-  let backgroundColor, hoverColor;
-  if (execState?.status === ExecutionStatus.Succeeded) {
-    backgroundColor = selected
-      ? theme.palette.DarkSuccessMain50
-      : theme.palette.DarkSuccessMain;
-    hoverColor = theme.palette.DarkSuccessMain75;
+  let status = execState?.status;
+  if (
+    execState?.status === ExecutionStatus.Failed &&
+    execState.failure_type == FailureType.UserNonFatal
+  ) {
+    status = ExecutionStatus.Warning;
+  }
 
-    // Warning color for non-fatal errors.
+  let backgroundColor;
+  if (execState?.status === ExecutionStatus.Succeeded) {
+    backgroundColor = theme.palette.green[100];
   } else if (
     execState?.status === ExecutionStatus.Failed &&
     execState.failure_type == FailureType.UserNonFatal
   ) {
-    backgroundColor = selected
-      ? theme.palette.DarkWarningMain50
-      : theme.palette.DarkWarningMain;
-    hoverColor = theme.palette.DarkWarningMain75;
+    backgroundColor = theme.palette.yellow[100];
   } else if (execState?.status === ExecutionStatus.Failed) {
-    backgroundColor = selected
-      ? theme.palette.DarkErrorMain50
-      : theme.palette.DarkErrorMain;
-    hoverColor = theme.palette.DarkErrorMain75;
+    backgroundColor = theme.palette.red[100];
   } else if (execState?.status === ExecutionStatus.Canceled) {
-    backgroundColor = selected ? 'gray.700' : 'gray.500';
-    hoverColor = 'gray.600';
+    backgroundColor = theme.palette.gray[200];
   } else if (execState?.status === ExecutionStatus.Pending) {
-    backgroundColor = selected ? 'blue.300' : 'blue.100';
-    hoverColor = 'blue.200';
+    backgroundColor = theme.palette.gray[200];
   }
 
+  const statusIndicatorComponent = (
+    <Box
+      sx={{
+        display: 'flex',
+        alignItems: 'center',
+        backgroundColor: backgroundColor,
+        // Even though the BaseNode's border radius is 8px, the differing dimensions
```

Comment:
This is odd, curious if there are any wrapping margins/paddings that we may need to unset?

Reply:
The inside boxes don't have any margin/paddings. I'll create a follow-up task to look into this, but it doesn't show any gaps right now.